pkg/metrics/metrics.go

```diff
@@ -85,3 +107,12 @@ func (metrics *metrics) ObserveDurationByState(state *api.StateItem) error {
 	metrics.Duration.With(prometheus.Labels{"state": string(state.State)}).Observe(duration.Seconds())
 	return nil
 }
+
+func (metrics *metrics) ObserveUpdateDurationByType(typ string, duration time.Duration) {
```

Comment:
```suggestion
func (metrics *metrics) ObserveUpdateDurationByType(type string, duration time.Duration) {
```

Reply:
It is not a typo. type is a keyword in go and cannot be used as variable name.